conans/client/generators/markdown.py

```diff
@@ -35,149 +44,187 @@
     {%- endmacro %}
 """)
 
-generator_cmake_tpl = textwrap.dedent("""
-    ### Generator ``cmake``
+buildsystem_cmake_tpl = textwrap.dedent("""
+    ### CMake
 
-    Add these lines to your *CMakeLists.txt*:
+    #### Generator [CMakeToolchain](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html)
+    `CMakeToolchain` is the toolchain generator for CMake. It will generate a toolchain
+    file that can be used in the command-line invocation of CMake with
+    `-DCMAKE_TOOLCHAIN_FILE=conantoolchain.cmake`. This generator translates the current
+    package configuration, settings, and options, into CMake toolchain syntax.
+
+    #### Generator [CMakeDeps](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html)
+    The `CMakeDeps` helper will generate one `xxxx-config.cmake` file per dependency,
+    together with other necessary `.cmake` files like version, flags, and directory data
+    or configuration.
+
+    If you use it together with the `CMakeToolchain`, these `xxx-config.cmake` files will be found when you run a `find_package()` in your `CMakeLists.txt` file:
+
+    ```
+    find_package({{ cmake_variables.file_name }})
 
-    ```cmake
-    include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-    conan_basic_setup(TARGETS)
+    # Use the global target
+    target_link_libraries(<target_name> {{ cmake_variables.global_target_name }})
 
-    target_link_libraries(<library_name> CONAN_PKG::{{ cpp_info.get_name("cmake") }})
+    {% if requirement.cpp_info.components is iterable and requirement.cpp_info.components %}
```

Comment:
I think this is a better approach to check empty lists:

```suggestion
    {% if requirement.cpp_info.components is defined and requirement.cpp_info.components|length > 0  %}
```

Reply:
Even I think it should be enough to put:

 ```python
{% if requirement.cpp_info.components|length  %}
```